Homework10: Ekaterina Zavadskaya

KateZavadskaya opened the pull request:
Add done homework10

Docs: hm10_1.py - hm10_3.py

KateZavadskaya committed:
Feat(hm10): add empty files for hm10

Docs: hm10_1.py - hm10_3.py

KateZavadskaya committed:
Feat(hm10): add done hm10

Add done hm10 with 3 files

Docs: hm10_1.py - hm10_3.py

Stml89 commented on hm10_1.py:
```diff
@@ -0,0 +1,37 @@
+                    raise ValueError(f"{i}. {y} Число меньше 0")
+            else:
+                raise ValueError(f"{i}. {y} Невозможно сравнить")
+                # i += 1
```
remove it

Stml89 commented on hm10_1.py:
```diff
@@ -0,0 +1,37 @@
+
+        lst = []
+        for x in args:
+            lst.append(x)
```
work with args, you do not need to copy these values to list()

Stml89 commented on hm10_1.py:
```diff
@@ -0,0 +1,37 @@
+            lst.append(x)
+        for i, y in enumerate(lst, start=1):
+            if isinstance(y, int):
+                if y > 0:
+                    print(f"{i}. {y} Число больше 0")
+                elif y == 0:
+                    raise ValueError(f"{i}. {y} Число равно 0")
+                elif y < 0:
+                    raise ValueError(f"{i}. {y} Число меньше 0")
+            else:
+                raise ValueError(f"{i}. {y} Невозможно сравнить")
+                # i += 1
```
Positive value, value less the 0

```suggestion
                if y < 0:
                    raise ValueError(f"{i}. {y} Невозможно сравнить")
```

KateZavadskaya replied:
Deleted exrta code and used args, ok
Do not use your advice:
1. it is easier to understand the numeric of the value
2. we need to understand what we have: number or not. And only then compare

Stml89 replied:
Regarding `isinstance` agreed, but what about to change `if-elif-elif` to just 1 `if`?

KateZavadskaya replied:
Mmm, the task is to lavidate the argument: is it positive. If not positive - tell it. 0 - not positive, not negative. If make "Positive value, value less the 0", what about 0 and not numeric/float.
Simple is better than complex, logical is better than confusing :) For me my logic is more logical :)

Stml89 commented on hm10_2.py:
```diff
@@ -0,0 +1,32 @@
+
+        """ Function takes arguments in any value """
+
+        lst = []
+        for x in args:
+            lst.append(x)
```
use `args`

Stml89 commented on hm10_2.py:
```diff
@@ -0,0 +1,32 @@
+    return a
+
+
+my_func(1, 9.1, "k")
```
`assert ...`

KateZavadskaya replied:
Used args, ok
I don’t understand how to use assert correctly, as it duplicates information. Please, help with the correct use with assert without duplicates information

Stml89 commented on hm10_3.py:
```diff
@@ -0,0 +1,56 @@
+
+assert add("9", 9) == "99", "Error processing (str, int)"
+assert add(9, 9) == "99", "Error processing (str, int)"
+assert add("9", "9") == "99", "Error processing (str, int)"
```
what if the count of arguments is 3?

KateZavadskaya replied:
Chanched arguments for 3 in first variant (not with #), ok.
But I understand, it also could be 5, 100, 100500. And I do not understand, how to fix it. 
If it needs 3 - done
If it needs 5, 100, 100500 - and as for me it should be so - I do not nderstand the correct fix, please help

Stml89 replied:
Could you please rephrase your question , I'm not sure that understood it

KateZavadskaya replied:
I do not understand is it difference between 3 arguments and indefinite quantity. 
Now I fix it for 3 arguments, but you can ask for 4, 7, 100 arguments.  In that cause, should I add the 4th agrument or my cod is incorrect? If it is incorrect - where is the mistake, I can not understand.

Stml89 replied:
In general, your solution will work with different counts of arguments. It uses `*args ` and`for`.
No need to write additional cases

KateZavadskaya committed:
Fix(code): create fix according comments

hm10_1.py: remove extra code - instead of list() use args
hm10_2.py: remove exrta code - instead of list() use args
hm10_3.py: create code for 4 arguments

Docs: hm10_1.py - hm10_3.py

Stml89 commented on hm10_3.py:
```diff
@@ -0,0 +1,61 @@
+    return decorator
+
+
+@typed(str, int, str)
```
the requirements say that only 1 type must be passed
```suggestion
@typed(str) / @typed(int) / ....
```

KateZavadskaya committed:
Fix(code): add fix to hm10_3.py

Fix type argument

Doc:hm10_3.py
